api/src/main/java/uk/gov/dvsa/mot/app/MotrRequestHandler.java

```diff
@@ -0,0 +1,252 @@
+import com.amazonaws.serverless.proxy.model.ApiGatewayRequestContext;
+import com.google.common.base.Strings;
+import com.google.inject.Inject;
+
+import org.apache.log4j.Logger;
+
+import uk.gov.dvsa.mot.persist.model.DvlaVehicle;
+import uk.gov.dvsa.mot.trade.api.BadRequestException;
+import uk.gov.dvsa.mot.trade.api.InternalServerErrorException;
+import uk.gov.dvsa.mot.trade.api.InvalidResourceException;
+import uk.gov.dvsa.mot.trade.api.MotrResponse;
+import uk.gov.dvsa.mot.trade.api.TradeException;
+import uk.gov.dvsa.mot.trade.read.core.MotrReadService;
+import uk.gov.dvsa.mot.vehicle.hgv.HgvVehicleProvider;
+import uk.gov.dvsa.mot.vehicle.hgv.model.HgvPsvVehicle;
+import uk.gov.dvsa.mot.vehicle.hgv.model.TestHistory;
+import uk.gov.dvsa.mot.vehicle.hgv.model.Vehicle;
+import uk.gov.dvsa.mot.vehicle.read.core.VehicleReadService;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.TemporalAdjusters;
+import java.util.Optional;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.Response;
+
+@Path("/")
+public class MotrRequestHandler extends AbstractRequestHandler {
+    private static final String HGV_VEHICLE_TYPE = "HGV";
+    private static final Logger logger = Logger.getLogger(MotrRequestHandler.class);
+
+    private MotrReadService motrReadService;
+    private VehicleReadService vehicleReadService;
+    private HgvVehicleProvider hgvVehicleProvider;
+    private String awsRequestId;
+
+    public MotrRequestHandler() {
+        super();
+    }
+
+    public MotrRequestHandler(boolean injectSelf) {
+        super(injectSelf);
+    }
+
+    @Inject
+    public void setVehicleReadService(VehicleReadService vehicleReadService) {
+        this.vehicleReadService = vehicleReadService;
+    }
+
+    @Inject
+    public void setHgvVehicleProvider(HgvVehicleProvider hgvVehicleProvider) {
+        this.hgvVehicleProvider = hgvVehicleProvider;
+    }
+
+    @Inject
+    public void setMotrReadService(MotrReadService motrReadService) {
+        this.motrReadService = motrReadService;
+    }
+
+    @GET
+    @Path("motr/v2/search/registration/{registration}")
+    @Produces("application/json")
+    public Response getVehicle(@PathParam("registration") String registration,
+                               ContainerRequestContext requestContext) throws TradeException {
+        try {
+
+            ApiGatewayRequestContext context = (ApiGatewayRequestContext) requestContext.getProperty(
+                    RequestReader.API_GATEWAY_CONTEXT_PROPERTY);
+            if (context != null) {
+                awsRequestId = context.getRequestId();
+            }
+
+            logger.info(String.format("Entering MotrRequestHandler, awsRequestId: %s", awsRequestId));
+
+            if (Strings.isNullOrEmpty(registration)) {
```

Comment:
are we sure that that method will be entered with registration as a path param will not be there ?